### ISSUE TITLE
Do we want to compile warning free?

### ISSUE THREAD

**Issue by MathiasMagnus** [keryell/triSYCL#19]
I know I have a fetish for writing warning-free, cross-platform, cross-compiler C++ code, but it might not be a goal. It does bug me though that the console literally explodes when compiling the unit tests.

MSVC cries over it's Debug Iterators a lot inside the SYCL short vector classes, GCC keeps whining about deprecated function declarations referred to inside stock Ubuntu CL/cl.h and Clang... for just about everything else.

It is very hard to test a new feature of SYCL, whether it provokes even just a warning in any of the tests, when the console is just beyond being of any use.

Should there be some aim like: GCC 6+, Clang 4+, VS2017+, all with -std=c++1z or /std:latest be warning-free?

**Comment by keryell** [comment]
Yes, it would be nice to have a warning-free compilation. It was the case up to some recent combination of Boost.Compute and GCC 6...
I have not investigated but it is painful when experimenting... :-(

**Comment by JoanThibault** [comment]
On my machine (Ubuntu 16.04, GCC 5.4) triSYCL compiles warning-free

**Comment by keryell** [comment]
I think the warning havoc started when I switched to Ubuntu 16.10 & GCC 6.2 with Boost.Compute, when compiling in OpenCL mode.

**Comment by MathiasMagnus** [comment]
#29
